src/sprite_clone.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * This file is part of EasyRPG Player.
+ *
+ * EasyRPG Player is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * EasyRPG Player is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with EasyRPG Player. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// Headers
+#include "graphics.h"
+#include "sprite_clone.h"
+#include "output.h"
+
+// Constructor
+SpriteClone::SpriteClone(Sprite* nreference, int nx, int ny, int nz) : reference(nreference), type(TypeClone), x(nx), y(ny), z(nz) {
+	if (reference == nullptr)
+		Output::Warning("Drawable reference null, it won't draw anything.");
+}
```

Comment:
imo you can pass the shared ptr of the Sprite around (typedef'd as SpriteRef). Raw pointers always have to risk to be dangling and crash.
The reference == nullptr warning-check can be removed imo.